compiler/src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -791,8 +791,8 @@ class DottyBackendInterface(outputDirectory: AbstractFile, val superCallsMap: Ma
     }
     def methodSymbols: List[Symbol] =
       for (f <- toDenot(sym).info.decls.toList if f.isMethod && f.isTerm && !f.isModule) yield f
-    def serialVUID: Option[Long] = None
-
+    def serialVUID: Option[Long] =
+      sym.getAnnotation(defn.SerialVersionUIDAnnot).flatMap(_.argumentConstant(0).map(_.longValue))
```

Comment:
You can pass a non constant argument to SerialVersionUID and it will still compile, which silently stops the field from being emitted, e.g.
```scala
@SerialVersionUID(A.id)
class A extends Serializable

object A extends App {
  def id = 42L
}
```

Reply:
Indeed, fixed.